### PR TITLE
[backends] Add userbenchmark for Torch-TRT Backend

### PR DESCRIPTION
**Pull request by gs-olive**
- Add handling for benchmarking multiple models in a single run using `ModelTask` instances
- Add error catching for testing all models and producing metrics
- Add CI yaml file to specify run configuration

### CODE REVIEW

File: userbenchmark/torch_trt/run.py [hunk 1]
```diff
@@ -8,14 +8,20 @@
...
+from torchbenchmark.util.experiment.instantiator import (
+    TorchBenchModelConfig,
+    load_model_isolated,
+)
 from torchbenchmark import (
+    ModelTask,
     load_canary_model_by_name,
     load_model_by_name,
-    list_models,
+    _list_model_paths,
```
Comment (xuzhao9) [inline userbenchmark/torch_trt/run.py:21]: It is better to use the `list_models()` API from `torchbenchmark.util.experiment.instantiator`. functions starting with `_` are not supposed to be used by userbenchmarks.

**Review by xuzhao9 — APPROVED** [review]
LGTM overall, see nits in the inline comment.

File: userbenchmark/torch_trt/run.py (hunk 1 above)
Comment (xuzhao9) [reply userbenchmark/torch_trt/run.py:21]: @gs-olive can you help address this so that we can merge this PR?

Comment (gs-olive) [reply userbenchmark/torch_trt/run.py:21]: Thank you for the review, and apologies for the delay on this. I will raise another PR to address this issue and clean it up, if that would be preferable? Additionally, I was wondering how I could monitor the CI runs for this userbenchmark?

Comment (gs-olive) [reply userbenchmark/torch_trt/run.py:21]: Raised #1821 addressing this comment

**Pull request closed without merging**